### ISSUE TITLE
neomutt, neovim and many others feeds do not showing

### ISSUE THREAD

**Issue by vvbalashoff** [newsboat/newsboat#2820]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
System: Linux 6.1.0-23-amd64 (x86_64)
Compiler: g++ 12.2.0
ncurses: ncurses 6.4.20221231 (compiled with 6.4)
libcurl: libcurl/7.88.1 GnuTLS/3.7.9 zlib/1.2.13 brotli/1.0.9 zstd/1.5.4 libidn2/2.3.3 libpsl/0.21.2 (+libidn2/2.3.3) libssh2/1.10.0 nghttp2/1.52.0 librtmp/2.3 OpenLDAP/2.5.13 (compiled with 7.87.0)
SQLite: 3.40.1 (compiled with 3.40.1)
libxml2: compiled with 2.9.14

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):
```
auto-reload yes       # автоматически запускать поток общего обновления                                                                
reload-time 5         # обновлять все каждые 5 минут                                                                                   
confirm-exit yes      # всегда спрашивать пользователя, действительно ли он хочет выйти                                                

notify-always no                                                                                                                       
browser "xdg-open '%u'                                                                                                                 
keep-articles-days 40                                                                                                                  

# bind-keys                                                                                                                            
bind-key LEFT prev-unread
bind-key RIGHT next-unread
bind-key щ open-in-browser

color background color244 color234 # Generic Background
color article    white color234
color listnormal color250 color234 #Color the listview items
color listfocus  color250 color239 #Selected listveiew item.
color info       color137 color235
color article    color250 color234 #Text/background when reading an article
color listnormal_unread color250 color234 
color listfocus_unread color250 color239
```

Steps to reproduce the issue:

1. add to urls: https://neomutt.org/feed.xml or https://neovim.io/news.xml

2. wait feed to update.

3. there are no feeds in list

**Other info you think is relevant**:
May be its irrelevant, but anyway.
I'm using cron script to download feeds to local files. and my urls contents something like:
```
file:///home/vvb/.newsboat/rss/neovim.io.xml                                                                                           
file:///home/vvb/.newsboat/rss/neomutt.org.xml
```
So, I can read those downloaded xmls. They have header:
```
<?xml version="1.0" encoding="UTF-8"?>                                                                                                 
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
```
I have about 60 other feeds and some of them are working right, but some are not. I think the problem is in xml parser somewhere in newsboat.

**Comment by dennisschagt** [comment]
Looks like the `newsboat -v` copy-paste is missing the actual newsboat version line (normally printend above the `System: Linux 6.1.0-23-amd64 (x86_64)` line).
Which version of newsboat are you using?

I've tried to reproduce on a few versions, somewhat randomly chosen: r2.32, r2.36, and current master.
For me both of the feeds shown at 1. load without issue.
However, I did not apply your config.
Now that I'm looking at it, I see the `keep-articles-days 40` line.
In case of the mentioned feeds, the latest articles are older than 40 days (2024-04-25 and 2024-05-16), so it is expected that all of their articles with a date longer than 40 days ago are dropped.

Did you expect different behavior from the `keep-articles-days` config option?
